lib/models/requestResponse.ts

```diff
@@ -1,23 +1,28 @@
 'use strict';
 
-const assert = require('../utils/assertType');
+import * as assert from '../utils/assertType';
+import { JSONObject } from '../utils/interfaces';
+import { KuzzleError } from '../errors/kuzzleError';
 
-const
-  _request = 'request\u200b',
-  _headers = 'headers\u200b';
+const _request = 'request\u200b';
+const _headers = 'headers\u200b';
 
 class Headers {
+  public headers: JSONObject;
+  private namesMap: Map<string, string>;
+  private proxy: JSONObject;
+
   constructor() {
     this.namesMap = new Map();
     this.headers = {};
     this.proxy = new Proxy(this.headers, {
-      get: (target, name) => this.getHeader(name),
-      set: (target, name, value) => this.setHeader(name, value),
-      deleteProperty: (target, name) => this.removeHeader(name)
+      get: (target, name) => this.getHeader(name as string),
+      set: (target, name, value) => this.setHeader(name as string, value),
+      deleteProperty: (target, name) => this.removeHeader(name as string)
     });
   }
 
-  getHeader (name) {
+  getHeader (name: string): string | undefined {
```

Comment:
I'm fairly new to typescript, but shouldn't the `name` property type defined as `string|symbol`?

Reply:
Can we have header with `symbol` ?

Reply:
I don't remember why we have that, but the code handles it. 

The way I see it, iou can  remove the handling of symbol parameters if you're sure we don't need it. Otherwise, you have to add `symbol` as a possible type for the name argument.